addons/account_facturx/models/account_move.py

```diff
@@ -222,15 +222,7 @@ def message_post(self, **kwargs):
         res = super(AccountMove, self).message_post(**kwargs)
 
         if 'no_new_invoice' not in self.env.context and len(self) == 1 and self.state == 'draft':
-            # Get attachments.
-            # - 'attachments' is a namedtuple defined in mail.thread looking like:
-            # _Attachment = namedtuple('Attachment', ('fname', 'content', 'info'))
-            # - 'attachment_ids' is a list of ir.attachment records ids.
-            attachments = kwargs.get('attachments', [])
-            if kwargs.get('attachment_ids'):
-                attachments += self.env['ir.attachment'].browse(kwargs['attachment_ids'])
-
-            for attachment in attachments:
+            for attachment in res.attachment_ids:
```

Comment:
@oco-odoo 
actually this was meant to fix issue mentioned in doc as `When trying to post the tax entry`, see related traceback https://pastebin.com/DZ7Dwitm , but I find csn's proposed fix(https://github.com/odoo/enterprise/pull/5402) is the actual fix for this specific issue. However, current fix is still valid because it fixes the root problem. When an invoice is to be generated through email(by configuring journal alias and fetching email from that alias), similar traceback would be faced. Reason is, `attachments` is a tuple, which is passed as param to `_create_invoice_from_attachment`, which expects attachment recordset. However, `attachments` and `attachment_ids` param will be processed by `_message_post_process_attachments`(called from super's `message_post`) and will generate attachments and set it them `attachment_ids` field, so I just used it

Reply:
Alright. Actually, we need to do both.